core/Plugin/Manager.php

```diff
@@ -1374,4 +1377,24 @@ public function loadPluginTranslations()
             $translator->addDirectory(self::getPluginsDirectory() . $pluginName . '/lang');
         }
     }
+
+    /**
+     * @param $plugin Plugin
+     * @throws \Exception
+     */
+    private function throwIfPluginMissingDependencies($plugin)
```

Comment:
Can you maybe move this method in another class and make it public so it can be tested? the string formatting part might be also useful for plugins/CorePluginsAdmin/templates/macros.twig and https://github.com/piwik/piwik/blob/2.16.2/core/Plugin/Manager.php#L831 where it should be actually deactivated the plugin even before we try to install I think